app/assets/javascripts/discourse/app/components/user-status-picker.js

```diff
@@ -0,0 +1,50 @@
+import Component from "@ember/component";
+import { action, computed } from "@ember/object";
+import { scheduleOnce } from "@ember/runloop";
+import { emojiUnescape } from "discourse/lib/text";
+import { escapeExpression } from "discourse/lib/utilities";
+
+export default class UserStatusPicker extends Component {
+  isFocused = false;
+  emojiPickerIsActive = false;
+  emoji = null;
+  description = null;
+
+  @computed("emoji")
+  get emojiHtml() {
+    const emoji = escapeExpression(`:${this.emoji}:`);
+    return emojiUnescape(emoji);
+  }
+
+  @action
+  blur() {
+    this.set("isFocused", false);
+  }
+
+  @action
+  emojiSelected(emoji) {
+    this.set("emoji", emoji);
+    this.set("emojiPickerIsActive", false);
+
+    scheduleOnce("afterRender", () => {
+      document.querySelector(".btn-emoji").focus();
```

Comment:
It's usually safer to do `document.querySelector(".btn-emoji")?.focus();`

Reply:
I've made it safer as you suggested, but now have some doubts. Like with the previous comment, not sure if we have to be so protective.

The component is designed in a way that it always have an emoji button. Let's say we set focus without `?`:
```javascript
document.querySelector(".btn-emoji").focus()
```

If somebody removes `btn-emoji` from the component, tests will fail, they will see that and remove the code for setting focus.

But if we set focus with `?` and somebody removes `btn-emoji`, tests will be green. Chances are that the code for setting focus will remain in the code base doing nothing.

In a way, by handling a possible null value in the code we're saying "the null value can appear here, if it happens that's fine". But that isn't the case here. If `.btn-emoji` was removed, this code should be removed as well. An explicit exception may be better than silent swallowing of the problem.

Again, not strongly against adding `?` (I've already added it). Curious, what do you think?